File: tools-manage/invocationlab-admin/src/main/java/org/javamaster/invocationlab/admin/enums/ProjectType.java
package org.javamaster.invocationlab.admin.enums;

/**
 * @author yudong
 * @date 2022/11/8
 */
public enum ProjectType {
    PERSONAL(1),
    GROUP(2),
    ;

    public final int type;

    ProjectType(int type) {
        this.type = type;
    }

    public static ProjectType getByType(Integer type) {
        if (type == null) {
            return GROUP;
        }
        for (ProjectType value : ProjectType.values()) {
            if (value.type == type) {
                return value;
            }
        }
        return GROUP;
    }
}

File: tools-manage/invocationlab-admin-21/src/main/kotlin/org/javamaster/invocationlab/admin/enums/ProjectType.kt
package org.javamaster.invocationlab.admin.enums

/**
 * @author yudong
 * @date 2022/11/8
 */
enum class ProjectType(@JvmField val type: Int) {
    PERSONAL(1),
    GROUP(2),
    ;

    companion object {
        @JvmStatic
        fun getByType(type: Int?): ProjectType {
            if (type == null) {
                return GROUP
            }
            for (value in entries) {
                if (value.type == type) {
                    return value
                }
            }
            return GROUP
        }
    }
}
